app/javascript/components/lease_ups/LeaseUpApplicationsPage.js

```diff
@@ -1,39 +1,81 @@
 import React from 'react'
-import { flow } from 'lodash'
+import { flow, isEmpty } from 'lodash'
 
 import LeaseUpApplicationsTableContainer from './LeaseUpApplicationsTableContainer'
 import TableLayout from '../layouts/TableLayout'
-import mapProps from '~/utils/mapProps'
 import { mapListing, mapApplicationPreference } from '~/components/mappers/soqlToDomain'
 import { buildLeaseUpModel } from './leaseUpModel'
 import appPaths from '~/utils/appPaths'
+import apiService from '~/apiService'
+import EagerPagination from '~/utils/EagerPagination'
+import Context from './context'
+import mapProps from '~/utils/mapProps'
+
+class LeaseUpApplicationsPage extends React.Component {
+  state = { loading: false, applications: [], pages: 0 }
+
+  constructor (props) {
+    super(props)
+    this.eagerPagination = new EagerPagination(20, 100)
+  }
+
+  fetchApplications = async (page) => {
+    const response = await apiService.fetchLeaseUpApplications(this.props.listing.id, page)
+    return {
+      records: !isEmpty(response.records) ? response.records.map(flow(mapApplicationPreference, buildLeaseUpModel)) : [],
```

Comment:
if you use lodash map, you do not have worry about checking if the value is empty or null, it returns an array always: `map(null, someFunction) => []`

so you can do:
`map(response.records, flow(mapApplicationPreference, buildLeaseUpModel))`